.github/workflows/python-ci.yml

```diff
@@ -0,0 +1,47 @@
+name: Python CI
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  code-quality:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        token: ${{ secrets.GH_PAT }}
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.x'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install black pyright pytest
```

Comment:
Oh actually I forgot to install `pytest-mock` (it allows you to use mockers from `unittest`, which creates fake object/return values), could you add it in here real quick and do `poetry add pytest-mock` as well? Thanks :D

Reply:
You don't need to run anything different, we just need to install it for `pip` in this script and `poetry` for the whole project

Reply:
so change it to:
``pip install black pyright pytest pytest-mock``
and under that add:
``poetry add pytest-mock`` ?

Reply:
Can just alter the pyproject.toml file to be:

[tool.poetry.dev-dependencies]
pytest = "^7.3.1"
black = "^23.3.0"
pyright = "^1.1.316"
pytest-mock = "^3.x"

as well right?

Reply:
> so change it to: `pip install black pyright pytest pytest-mock`
> 
> and in the project dir run: `poetry add pytest-mock --dev` ?

Yep, don't think you'll need the `--dev` for poetry?

Reply:
Yep, took a while but got it 👍